apps/sms/views/conversation/js/attachment.js

```diff
@@ -73,32 +73,37 @@
       //   - Blob mimetype is unsupported but file extension is valid.
       //   - File extenion is missing or invalid but mimetype is supported.
 
-      var mimetype =
-        MimeMapper.guessTypeFromFileProperties(
-          this.name,
-          this.blob.type.toLowerCase()
-        );
-      var filename = MimeMapper.ensureFilenameMatchesType(this.name, mimetype);
-
-      // Override filename, so that every attachment that is saved via "open"
-      // activity will be placed in the single location.
-      filename = ATTACHMENT_FOLDER_PATH + getBaseName(filename);
-
-      var activity = new MozActivity({
-        name: 'open',
-        data: {
-          type: mimetype,
-          filename: filename,
-          blob: this.blob,
-          allowSave: options && options.allowSave
-        }
+      navigator.mozL10n.formatValue('unnamed-attachment').then(unnamed => {
```

Comment:
Thanks! I'll update the patch but will not go for transforming the promise chain here to minimize the size of the patch.